xdggs/index.py

```diff
@@ -38,7 +38,9 @@ def from_variables(
         _, var, _ = _extract_cell_id_variable(variables)
 
         grid_name = var.attrs["grid_name"]
-        cls = GRID_REGISTRY[grid_name]
+        cls = GRID_REGISTRY.get(grid_name)
```

Comment:
If we publicly expose the `DGGSInfo` classes, we should probably also support passing grid parameters either as separate kwargs or as an instance of a `DGGSInfo` subclass passed via a `grid_info` kwarg.

Reply:
good point. I'd probably go with the latter. However, I don't think we need to do this in this in this PR.